themes/default/content/docs/intro/concepts/inputs-outputs.md

```diff
@@ -651,6 +651,327 @@ variables:
+                "Next": "World",
+            },
+            "World": {
+                "Type": "Task",
+                "Resource": worldFunction.arn,
+                "End": true,
+            },
+        },
+    })
+});
+```
+
+### Example Parsing JSON Outputs to Objects
+
+The following is an example of Pulumi code that operates on a JSON object and removes all of the policy statements by replacing it with an empty list.
+
+```typescript
+const policyWithNoStatements = pulumi.jsonParse(jsonIAMPolicy).apply(policy => {
+    // delete the policy statements
+    policy.Statement = [];
+    return policy;
+});
+```
+
+For more information view the NodeJS documentation.
+
+{{% /choosable %}}
+{{% choosable language python %}}
+
+### Example Converting Outputs to JSON
+
+The following example allows you to natively represent the definition of an AWS Step Function State Machine and easily embed outputs from other resources in your stack.
+
+```python
+state_machine = aws_native.stepfunctions.StateMachine("stateMachine",
+    role_arn=sfn_role.arn,
+    state_machine_type="EXPRESS",
+    definition_string=pulumi.Output.json_dumps({
+        "Comment": "A Hello World example of the Amazon States Language using two AWS Lambda Functions",
+        "StartAt": "Hello",
+        "States": {
+            "Hello": {
+                "Type": "Task",
+                "Resource": hello_function.arn,
+                "Next": "World",
+            },
+            "World": {
+                "Type": "Task",
+                "Resource": world_function.arn,
+                "End": True,
+            },
+        },
+    })
+)
+```
+
+### Example Parsing JSON Outputs to Objects
+
+The following is an example of Pulumi code that operates on a JSON object and removes all of the policy statements by replacing it with an empty list.
+
+```python
+json_IAM_policy = pulumi.Output.from_input('''
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "VisualEditor0",
+            "Effect": "Allow",
+            "Action": [
+                "s3:ListAllMyBuckets",
+                "s3:GetBucketLocation"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "VisualEditor1",
+            "Effect": "Allow",
+            "Action": "s3:*",
+            "Resource": "arn:aws:s3:::my-bucket"
+        }
+    ]
+}
+''')
+
+def update_policy(policy):
+    # delete the policy statements
+    policy.update({'Statement': []})
+    return policy
+
+policy_with_no_statements = pulumi.Output.json_loads(json_IAM_policy).apply(update_policy)
+```
+
+For more details view the Python documentation.
+
+{{% /choosable %}}
+{{% choosable language go %}}
+
+### Example Converting Outputs to JSON
+
+{{% notes type="info" %}}
+The Pulumi Go SDK does not currently support Marshalling and Unmarshalling Maps with unknowns.
+
+This is being tracked [here](https://github.com/pulumi/pulumi/issues/12460).
+{{% /notes %}}
```

Comment:
@Frassle is this a good summary of the current state of Go?

Reply:
It's not just unknowns. Any nested outputs I think will trigger an error in Marshal because all output types are unmarshallable.